mpl_interactions/generic.py

```diff
@@ -480,3 +483,103 @@ def _onselect(self, verts):
 
     def _ipython_display_(self):
         display(self.fig.canvas)
+
+
+def hyperslicer(
+    arr,
+    cmap=None,
+    norm=None,
+    aspect=None,
+    interpolation=None,
+    alpha=None,
+    vmin=None,
+    vmax=None,
+    origin=None,
+    extent=None,
+    autoscale_cmap=True,
+    filternorm=True,
+    filterrad=4.0,
+    resample=None,
+    url=None,
+    ax=None,
+    slider_format_string=None,
+    title=None,
+    figsize=None,
+    display=True,
+    force_ipywidgets=False,
+    **kwargs,
+):
+
+    params = {}
+    ipympl = notebook_backend()
+    fig, ax = gogogo_figure(ipympl, figsize, ax)
+    use_ipywidgets = ipympl or force_ipywidgets
+    slider_format_strings = create_slider_format_dict(slider_format_string, use_ipywidgets)
+                            
+    name_to_dim = {}
+    slices = [0 for i in range(arr.ndim-2)]
+    #Just pass in an array - no kwargs
+    for i in range(arr.ndim - 2):
+        kwargs[f'axis{i}'] = np.arange(arr.shape[i]) 
+        slider_format_strings[f'axis{i}'] = '{:d}'
+        name_to_dim[f'axis{i}'] = i
+        
+    def update(change, label, key):
+        if label:
+        # continuous
+            params[key] = kwargs[key][change["new"]]
+            label.value = slider_format_strings[key].format(kwargs[key][change["new"]])
+        else:
+        # categorical
+            params[key] = change["new"]
+        if title is not None:
+            ax.set_title(title.format(**params))
+            
+        if arr.ndim>=2:
+            slices[name_to_dim[key]] = change['new']
+            new_data = np.asarray(arr[tuple(slices)])
+            im.set_data(new_data)
```

Comment:
1. no longer need this `asarray` thats important for imshow bc we don't know what the function provided by the user might return. But by this point we know for sure that we have a numpy array.

2.  We should check earlier in the function if the number of dimensions make sense. and then we won't need to check here

3. I think what you did to get the indices is fine. You could finagle your way into doing this with `params` but that doesn't seem worth it.
2. I think y